Add linting of player code

arokor opened the pull request:
- Adds linting to all player platforms. Run `yarn lint` to lint the code.
- Adds lint step in github CI test workflow for player platform code
- Fixes the lint issues. ***There are no functionality changes in this PR***. All changes are just lint-fixes.

manast commented on packages/platforms/android-player/src/ts/classes/android-machine.ts:
```diff
@@ -54,7 +54,7 @@ export class AndroidMachine implements Machine {
         longitude: location.coords.longitude,
       };
-    } catch (e) {
+    } catch {
```
wouldn't it be better to cast the e argument to Error and throw that instead?

arokor replied:
According to the machine interface we should return undefined if no location can be determined.

manast replied:
wouldn't we at least want to log out these errors? otherwise it would difficult to understand why it goes wrong when it goes wrong.

arokor replied:
Added logging of this error (for all platforms)

manast commented on packages/platforms/android-player/src/ts/classes/android-machine.ts:
```diff
@@ -67,7 +67,7 @@ export class AndroidMachine implements Machine {
     try {
       appInfo = await App.getInfo();
-    } catch (e) {
+    } catch {
```
wouldn't it be better to cast the e argument to Error and throw that instead?

arokor replied:
I think this was written like this to make the code run without the App plugin (when running the code locally in a web browser). This is handy during development since it saves the overhead of building the Android app every time something in the ts code is changed. I also didn't want to change any functionality in this PR since it is just about linting.

manast reviewed (commented):
I see a couple of things, some catch clauses have eliminated the error object which I think it makes sense to keep so that we do not loose information of the error such as the stack trace and message. The other thing is if it is possible to move the lint dependencies to the root package.json instead of having them in every package, as that would be more cumbersome to maintain and keep up to date?

arokor commented:
> I see a couple of things, some catch clauses have eliminated the error object which I think it makes sense to keep so that we do not loose information of the error such as the stack trace and message. The other thing is if it is possible to move the lint dependencies to the root package.json instead of having them in every package, as that would be more cumbersome to maintain and keep up to date?

Moved the lint dependencies to root and rebased. Also added logging in some catch clauses, notably the geolocation functions

manast reviewed (approved):
LGTM
